src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -274,6 +274,36 @@
      (fn [^Connection conn]
        (describe-table-fks* driver conn table db-name-or-nil)))))
 
+(defn describe-table-indexes
+  "Default implementation of [[metabase.driver/describe-table-indexes]] for SQL JDBC drivers. Uses JDBC DatabaseMetaData."
+  [driver db table]
+  (sql-jdbc.execute/do-with-connection-with-options
+   driver
+   db
+   nil
+   (fn [^Connection conn]
+     ;; https://docs.oracle.com/javase/8/docs/api/java/sql/DatabaseMetaData.html#getIndexInfo-java.lang.String-java.lang.String-java.lang.String-boolean-boolean-
+     (with-open [index-info-rs (.getIndexInfo (.getMetaData conn)
+                                              nil ;; category
+                                              (:schema table)
+                                              (:name table)
+                                              ;; when true, return only indices for unique values when
+                                              ;; false, return indices regardless of whether unique or not
+                                              false
+                                              ;; when true, result is allowed to reflect approximate or out of data
+                                              ;; values. when false, results are requested to be accurate
+                                              false)]
+
+       (-> (group-by :index_name (into []
+                                       (filter #(and (= (:type %) DatabaseMetaData/tableIndexOther) ;; we only care about other user defined index types
```

Comment:
Why exclude `tableIndexHashed` and `tableIndexClustered`? These are still types of indices that we might care about. 

See https://learn.microsoft.com/en-us/sql/relational-databases/indexes/clustered-and-nonclustered-indexes-described?view=sql-server-ver16 for clustered indices on SQL Server.

And https://www.postgresql.org/docs/current/indexes-types.html#INDEXES-TYPES-HASH for hashed indices on Postgres.

Reply:
right! I thought if you create an index of type hash, it's still a user index type but turns out it's not.

for clustered index on postgres, it requires an index to work on so I thought since we'll sync the original index already, we can ignore it. Turns out if you cluster a table with an index, that index will have type "clustered".

anw I synced all of them in https://github.com/metabase/metabase/pull/36462/commits/9b9292dc3f732f3536e7c16ee256a9057c7744c3 and added tests